src/db/mongo/users.ts

```diff
@@ -55,9 +58,34 @@ export const updateUser = async (user: User): Promise<void> => {
   if (user.email) {
     user.email = user.email.toLowerCase();
   }
+  if (!user.publicKeys) {
+    user.publicKeys = [];
+  }
   const { _id, ...userWithoutId } = user;
   const filter = _id ? { _id: new ObjectId(_id) } : { username: user.username };
   const options = { upsert: true };
   const collection = await connect(collectionName);
   await collection.updateOne(filter, { $set: userWithoutId }, options);
 };
+
+export const addPublicKey = async (username: string, publicKey: string): Promise<void> => {
+  const collection = await connect(collectionName);
+  await collection.updateOne(
+    { username: username.toLowerCase() },
+    { $addToSet: { publicKeys: publicKey } },
+  );
+};
+
+export const removePublicKey = async (username: string, publicKey: string): Promise<void> => {
+  const collection = await connect(collectionName);
+  await collection.updateOne(
+    { username: username.toLowerCase() },
+    { $pull: { publicKeys: publicKey } },
+  );
+};
+
+export const findUserBySSHKey = async function (sshKey: string): Promise<User | null> {
```

Comment:
Same comment as above, does this correctly handle two users with the same key, or other duplicate key scenarios?